hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -24,199 +24,356 @@
-            enabled, ensure that the raw data is written and read back.
+            Perform accesses in order to provoke program resolution error. Test both, Software
+            interface and Hardware interface. Related covergroup is error_cg.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: ecc
+      name: error_prog_type
       desc: '''
-            Randomly enable ECC for a randomly selected set of pages. Randomly corrupt a single bit
-            in the memory that is about to be read and ensure that the ECC works - the corrupted bit
-            should be fixed. Corrupt randomly either the data or the ECC bits. Randomly corrupt 2
-            bits in the same word and ensure that the read results in error. Ensure that pages with
-            ECC not enabled reads back corrupted data without any errors.
+            Perform accesses in order to provoke program type error. Test both, Software interface
+            and Hardware interface. Related covergroup is error_cg.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: error_flash_phy
+      desc: '''
+            Perform accesses in order to provoke native flash error. Test both, Software interface
+            and Hardware interface. Related covergroup is error_cg.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: error_shadow_update
+      desc: '''
+            Perform accesses in order to provoke shadow register update error. Related covergroup
+            is error_cg.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: read_ecc_metadata
+      name: error_integrity
       desc: '''
-            TBD - in futurem support for reading the metadata from the ECC bits may be supported -
-            this is a placeholder to verify that feature.
+            Perform accesses in order to provoke register integrity error on flash controller and
+            flash memory. Related covergroup is error_cg.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: error
+      name: error_lc
       desc: '''
-            This test will attempt to generate flash error in all possible ways:
-            1. Read / program / erase a protected chunk
-            2. Set num words to 0 or out of bounds on read or program
-            3. Inject double bit errors when ECC is enabled
-            4. Program or read out-of-bounds info partition
-            5. Erase whole bank with bank erase not enabled
-            6. Randomly wiggle scramble enable in the middle of an op and ensure that the FSM does
-            not get locked up
+            Perform accesses in order to provoke life cycle management interface error. Related
+            covergroup is error_cg.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: flash_macro_timing
+      name: error_storage
```

Comment:
i believe this one is just like the integrity.

@cindychip 
@weicaiyang

Reply:
This is shadow reg related. We can remove it and import `shadow_reg_errors_testplan.hjson`.

Reply:
Ok,removed and added shadow_reg_errors_testplan.hjson.